State[]  stateTrajectory
State x0
State xf